### PR TITLE
mistakes

### PR DESCRIPTION
**Pull request by suova**
1)At first glance, that it was impossible to write a character that represents single quote or composed character that containing double quote. -> At first glance, it was impossible to write a character that represents  a single quote or a composed character containing a double quote.  Дважды пропущен артикль,  указательное местоимение that в данном контексте не нужно дважды ---(4 ошибки)

2)signs ->characters если в данном случае имеются в виду символы 
---(1 ошибка)

3)All of them make up one character (if it is written into single quotes), or one of characters in compound symbol.  -> All of them make up a single character (if it is written in single quotes), or one of the characters in a  compound symbol. 
Если имеется в виду одиночный символ то это single плюс отсутствует артикль, еще два артикля пропущены  ---(4 ошибки)

4)We referred earlier to the functions in the right part of a sentence which are calculated somehow after a substitution of variables. -> We have referred earlier to the functions in the right part of a sentence which are calculated somehow after a substitution of variables. Грамматика, действие без указания на точное время в прошлом Present Perfect ---(1 ошибка)

5)Now it is time to specify, how exactly, because it is impossible to write effective programs and to execute debugging of programs on Refal-5λ without it. ->
Now it's time to clarify exactly how, because without this it is impossible to write effective programs and debug programs on Refal-5λ. Грамматически неправильное предложение + очень по-русски(how exactly->exactly how)----------(2 ошибки)

6)a current status -> the current status хранит же определенный а не любой--------(1 ошибка)

7)The status of computation is described in the form of the active expression – expression of the Refal language which contains activation brackets, but at the same time cannot contain variables.->The status of computation is described as the active expression-the Refal expression which contains activation brackets, but at the same time cannot contain variables
Правильно as  the не  in the form of the-----------(1 ошибка)

8) Each step is an execution of the following sequence of actions. -> Each step is the execution of the following sequence of actions. артикль определенный так как действие определенное ---------(1 ошибка)

9)Refal-maсhine finds in a view field the most left couple of activation brackets, such that doesn’t include other angle brackets in this call. ->The Refal machine finds the leftmost pair of activation brackets in the field of view, one that does not include other angled brackets in this call.  Определенный артикль, более корректная фраза( the leftmost pair ) ---(2 ошибки)

10)This section of a view field is called primary active sub-expression.->This section of a view field is called the primary active sub-expression. артикль--------(1 ошибка)

11) Function can be either written on Refal or built-in - > The function can either be written on Refal or built-in. артикль и неправильное построение предложения
 ---(2 ошибки)

12)Refal-maсhine observes what is on the right ->The Refal-maсhine observes what is on the right Артикль----(1 ошибка)

13)Refal-maсhine finds function name in the program field -> The Refal-maсhine finds a function name in the program field Артикль 2 раза --(2 ошибки)

14)If it is possible to choose such values of variables in the left part of the
   current sentence that it might become the function argument, then the
   point 5 is implemented. -> If it is possible to choose such values of variables in the left part of the current sentence that it might become the function argument, then the  point 5 has been implemented. Грамматика ---(1 ошибка)

15)f there are no sentences anymore, then the refal-machine
   stops with a error "recognition impossible".-> f there are no sentences anymore, then the refal-machine stops with the error "recognition impossible". Ошибка конкретная (артикль)---(1 ошибка)

16)Refal-machine inserts received expression into the view field to the place of the primary active subexpression. ->The Refal-machine inserts received expression into the view field at the place of the primary active subexpression. Предлог и артикль --(2 ошибки)

17)REFAL-5 doesn’t support empty functions, any function has to have
   at least one sentence->REFAL-5 doesn’t support empty functions, any function must to have  at least one sentence Обязательно - must--(1 ошибка)

18)the call of such function leads to a error -> the call of such function leads
   to the error Ошибка конкретная (артикль)---(1 ошибка)

19)In this case there will be created the program which at start will display the error message and then will come to the end. ->In this case the program will be created which at start will display the error message and then will come to the end. Неправильно построено предложение ---(1 ошибка)

20)Firstly, the primary active subexpression chooses the most left couple of activation brackets which isn’t containing in itself other activation brackets. ->Firstly, the primary active subexpression selects the leftmost pair of activation brackets that does not contain any other activation brackets. Корректное словосочетание, выбирает - более корректно select, грамматическая ошибка-время обычное настоящее выполняется всегда -----(3 ошибки)

21)It follows that Refal – is applicative language, and an order of function computations is accurately defined: from left to right. ->It follows that the Refal – is an applicative language, and the order of function computations is accurately defined: from left to right. 3 артикля  -----(3 ошибки)

22) Secondly, it immediately follows from semantics of a view field that Refal implements so-called optimization of a tail recursion.->
Secondly, it immediately follows from semantics of a view field that the Refal implements so-called optimization of a tail recursion. артикль ---(1 ошибка)

23)In many imperative programming languages the recursion is quite expensive — overhead costs on preservation of a context of calculations are needed – points where process of execution has to return after the completion of a tail call.->
In many imperative programming languages a recursion is quite expensive — it requires the overhead of maintaining the calculation context – points where process of execution has to return after the completion of a tail call. предложение очень по-русски---(1 ошибка)

24)Therefore in such languages you are not to use a recursion for the implementation of the cyclic repeating actions (particularly as in imperative languages for this purpose there are iteration statements).-> Therefore, in such languages, you should not use a recursion to implement cyclically repeated actions (especially since there are iterative operators for this in imperative languages). 
Нет запятых, грамматика( you are not to use -> you should not use), более красиво с точки зрения английского(to implement cyclically repeated actions), грамматическая ошибка в построении предложения(as in imperative languages for this purpose there are iteration statements)  ---(4 ошибки)

25)In Refal functions don’t need to keep a context of calculations for return of a
value as the function doing a recursive call comes to the end before the next
recursive call. -> The Refal functions do not need to keep a calculation context to return a
 value as the function that performs a recursive call is terminated before the next 
recursive call.  Артикль, более красивое предложение ( comes to the end - приходит к концу…. лучше is terminated(прекращается)) --(2 ошибки)

всего 44

### CODE REVIEW

File: docs/3-basics.en.md [hunk 1]
```diff
@@ -1266,23 +1266,27 @@ _cyclic processes_ and to use the term _iteration,_ etc.
...
+
+* Refal expression is called an _object expression_.It can contain only symbols 
+and round brackets. Accordingly, terms that make up an object expression are
+called an _object terms._ Function arguments can be only an object expression.
+* An expression that assembled from symbols, round and angled brackets is called
+an _active expression_ or a _ground expression._ The content of the field 
+of view can only be an active expression.
+* An expression made up of symbols, structural brackets , and variables 
+is called a _pattern expression_ or _pattern._ The left sentence 
+part is a pattern expression.
```
Comment (Mazdaywik) [inline docs/3-basics.en.md]: Эти правки портят форматирование и смысла не несут.

File: docs/3-basics.en.md [hunk 2]
```diff
@@ -1296,13 +1300,13 @@ All four types of expressions can be imaged as a figure:
...
-sentences. The reason for this is that there were variables. Pattern expression
-is a template that describes multiple object expressions. Result expression is
-template from which active expression is built.
+Object and active expressions exist only during the execution of a program 
+such as function arguments and the content of the view field (the content 
+of static boxes and stack could be added we will consider later.) While 
+pattern and result expressions exist only in source code as left and right
+parts of sentences. The reason for this is that there were variables. Pattern 
+expression is a template that describes multiple object expressions. Result 
+expression is template from which active expression is built.
```
Comment (Mazdaywik) [inline docs/3-basics.en.md]: Смысла в этих правках тоже нет.

**Comment by Mazdaywik** [comment]
```
git remote add main https://github.com/bmstu-iu9/refal-5-lambda
git fetch --all
git checkout master
git rebase -i main/master
```
Тут откроется текстовый редактор с таким текстом:
```
pick 0a345b6 mistakes in the last part
pick 961cee4 new mistakes
pick 45b01b6 new mistakes
pick 4121bbf Update 3-basics.en.md

# …
# …
```
Первую строчку надо убрать:
```
pick 961cee4 new mistakes
pick 45b01b6 new mistakes
pick 4121bbf Update 3-basics.en.md
```
и закрыть текстовый редактор.
```
git push -f
```

**Comment by suova** [comment]
Ready!

**Pull request merged**